### PR TITLE
remove subjects not in demographics

### LINKED ISSUE #9: Subjects with imaging but no demographics should be removed during merge
When I am merging a demographic and an imaging json file with bagelbids, subjects that exist only in the imaging json file and not in the demographic json file should be removed, so that none of their meta data ends up in the merged jsonld graph.

- [ ] A warning is displayed when subjects are removed and the warning includes the subject IDs for all removed subjects
- [ ] the bagelbids merge process does not fail when the subject ID lists are mismatched between the demographic and the imaging json files

This is related to 
- https://github.com/neurobagel/project/issues/40
- https://github.com/neurobagel/project/issues/41

### PR DESCRIPTION
**Pull request by surchs**
Fixes #9

### CODE REVIEW

File: bagelbids/tests/test_merge.py [hunk 1]
```diff
@@ -133,11 +134,39 @@ def test_merge_if_demo_has_additional_subjects(bids_json, demo_json_long):
...
-    with pytest.warns(UserWarning, match=r"99"):
+    with pytest.warns(Warning, match=r"99"):
         result = merge_json(bids_json, demo_json_long)
     assert result == target_json
 
 
 def test_merge_if_bids_has_additional_subjects(bids_json_long, demo_json):
-    with pytest.raises(NotImplementedError):
-        merge_json(bids_json_long, demo_json)
+    # If there are more subjects in the BIDS dataset than in the demographic file
```
Comment (jarmoza) [inline bagelbids/tests/test_merge.py:143]: @surchs I've read through `merge_json` and `merge_on_subject`. Can you explain how this assert does not fail every time? I don't see the return value of `merge_on_subject` matching `target_json`

Comment (jarmoza) [reply bagelbids/tests/test_merge.py:143]: 👍 Understood after today's walk through of the test and fixtures code.

**Review by jarmoza — APPROVED** [review]
@surchs This is good to go. Let's open a general issue about adding docstrings and comments to the bagelbids repo.

**Comment by surchs** [comment]
sounds good @jarmoza, tracking the comments here: https://github.com/neurobagel/bagelbids/issues/15

**Pull request merged**